### ISSUE TITLE
Getting script errors when executing the .hta file

### ISSUE THREAD

**Issue by ovizii** [LesFerch/WinSetView#19]
here are some samples:

![image](https://user-images.githubusercontent.com/837902/128989909-ce91df6f-64b5-4acf-81d0-d882d4d534a8.png)

![image](https://user-images.githubusercontent.com/837902/128989950-70b05dd8-59fd-4b4c-b6ba-0304661ffecf.png)

![image](https://user-images.githubusercontent.com/837902/128990006-2e27e867-602c-4238-be92-5c78e105e3ec.png)

**Comment by LesFerch** [comment]
Please let me know if this fixed the issue for you. Also please let me know if you have any corrections for the cs-CZ language. Thank you.

**Comment by LesFerch** [comment]
@ovizii Have you tried the updated script?

**Comment by ovizii** [comment]
sorry for the delay, I was offline. I just tried and this version doesn't open at all upon double clicking the .hta file.

I am using Win 10 with the latest updates. I manually tried right-click and "open with" then selected: "Internet Explorer" - it opens and asks:

![image](https://user-images.githubusercontent.com/837902/129191020-a1078300-c41d-455e-888e-ab7134f2cdcb.png)

I select: open and it then simply closes. Might just be my setup - no idea what is wrong.

**Comment by ovizii** [comment]
> Please let me know if this fixed the issue for you. Also please let me know if you have any corrections for the cs-CZ language. Thank you.

I'm not czech, what you see int he screenshot just seems to be the default language?

**Comment by ovizii** [comment]
nope, just Windows Defender. don't worry, lets see if anyone else reports anything similar. Feel free to close the ticket after a while if nobody reports any problem.

**Comment by LesFerch** [comment]
@ovizii I really want to get to he bottom of this and I can assure you that you're not seeing standard behavior. You may have corrupted settings or, at worst, some malware messing with things. I wouldn't ignore it. If you can stay with me, I'm sure we can try some simple tests to narrow things down. For example, please try running this one line script:
[https://github.com/LesFerch/SampleCode/blob/main/UserAgent/Oneliner.hta](url)
It should first display a small alert box showing the mode the HTA is running in, followed by a blank HTA window.

**Comment by LesFerch** [comment]
@ovizii Maybe you just had a locked mshta.exe process. Please try logout/login or a restart and try again. If the problem persists, please try the one line script from my previous comment. Looking forward to hearing back. Thanks.

**Comment by LesFerch** [comment]
@ovizii Please also ensure that HTA files are associated with mshta.exe, not iexplore.exe. That is, Right-click, Open with, Choose another app should be:
![image](https://user-images.githubusercontent.com/79026235/129451322-a9252511-20e9-4605-b09c-046872b8cb63.png)

**Comment by ovizii** [comment]
thanks for all your help, of course I appreciate your help, I was just backing out in my last reply because I didn't want to cause too much trouble but seeing that you are very willing to help lets see.

I was on holiday hence my delayed replies and am now working remotely. I was giving your tool a try via RDP on my remote machine so let me restart it and get back to you.

**Comment by ovizii** [comment]
`<script>alert(document.documentMode)</script>`
just opens a little windows saying: "5" - not sure if that helps?

**Comment by ovizii** [comment]
> @ovizii Maybe you just had a locked mshta.exe process. Please try logout/login or a restart and try again. If the problem persists, please try the one line script from my previous comment. Looking forward to hearing back. Thanks.

That solved it. I looked for the process and terminated it and then your tool ran perfectly. 

All solved! Thanks a lot for your help.

**Comment by LesFerch** [comment]
@ovizii Thank you so much for getting back to me. I'm so relieved to hear that the updated script worked properly. Also, thank you for running the one line script. The fact that it reported "5" indicates that you do not have a registry setting that forces HTAs to run in IE10 mode. That leaves the mystery of why the original script ran in IE10 mode on your computer. If I could bother you for one more test... The two line script below will typically report "7" but it may report "10" on your computer. Can you please run this script and let me know what it reports? Again, thank you so much for your time!
```
<!DOCTYPE html>
<script>alert(document.documentMode)</script>
```

**Comment by ovizii** [comment]
not sure if it helps but here you go. Not what you expected :-/
![image](https://user-images.githubusercontent.com/837902/129702066-b45c81d0-73d6-4211-8aa0-a73f79ecab1e.png)

**Comment by LesFerch** [comment]
@ovizii Thank you! I checked again and found that both IE=9 and IE=10 modes will cause the same errors with my script. So, we now know that any HTA script with only a <!DOCTYPE html> declaration at the tope runs in IE=9 mode by default on your machine (instead of the usual 7), but a completely bare-bones script runs in the expected IE=5 mode. I'll have to do some research and experimenting to try to find out what setting(s) will cause this, but at least it's just academic. Since I added the line in my script to force it to run in IE=7 mode, I don't have to worry about it being in the wrong mode. Thanks again. :-)
